### ISSUE TITLE
KeePassXC crashing on database open if compiled with llvm's libcxx and running wayland

### ISSUE THREAD

**Issue by 0xIO32** [keepassxreboot/keepassxc#11385]
## Overview
[TIP]:  # ( DO NOT include screenshots of your actual database! )
[NOTE]: # ( Give a BRIEF summary about your problem )

## Steps to Reproduce
[NOTE]: # ( Provide a simple set of steps to reproduce this bug. )
1. Compile keepassxc and all the libraries that keepassxc depends on with libcxx (from llvm, instead of libstdcxx from gcc. You cannot mix libraries that are compiled with different c++ standard libraries).
2. Open KeePassXC (ensure you are running a wayland session)
3. Try to open a database (enter a password)

## Expected Behavior
[NOTE]: # ( Tell us what you expected to happen )
The database should open if the credentials are correct

## Actual Behavior
[NOTE]: # ( Tell us what actually happens )
The application crashes with a segmentation fault

## Context
[NOTE]: # ( Give us any additional information you may have. )
gdb backtrace:
![image](https://github.com/user-attachments/assets/be22ebda-23b4-4e5b-bde1-5bde2864a9d6)

It doesn't crash inside valgrind.

It is probably something related to qt, but as it only happens with keepassxc, I opened the issue here.

[NOTE]: # ( Paste debug info from Help → About here )
KeePassXC - 2.7.9
Revision: 8f6dd13

[NOTE]: # ( Pick choices based on your environment )
Operating System: Gentoo Linux
Desktop Env: Hyprland (also tried using Sway)
Windowing System: Wayland (it doesn't crash on X11)

**Comment by droidmonkey** [comment]
As that stack trace does not start or end with keepassxc code, this is definitely external to our program. Technically the start of the stack trace from a "initiating" point of view is line number 19. Closing unless you can point to specific KeePassXC code cause.

**Comment by 0xIO32** [comment]
> As that stack trace does not start or end with keepassxc code, this is definitely external to our program. Technically the start of the stack trace from a "initiating" point of view is line number 19. Closing unless you can point to specific KeePassXC code cause.

I think I found the issue:
https://bugreports.qt.io/browse/QTBUG-78652

It has been caused in qt, because of a animated cursor. Meaning the database unlock caused the "busy" cursor, which was animated and then crashed the application.

This is fixed in Qt6, but not Qt5. ~~Meaning, this issue will persist for me until KeePassXC updates to Qt6.
A workarround would be to not hover over the application while unlocking.~~
Found a working patch online.
